lib/game.rb

```diff
@@ -18,19 +18,22 @@ def welcome
 
   def turn
     alert_current_player
-    console.print_message("Player #{@current_player}, please enter a number 1-9")
-    index = @console.retrieve_user_input.to_i - 1
-    board.position_taken?(index) ? turn : @board.mark(index, @current_player)
+    retrieve_user_move
     console.print_message(@board.display)
     switch_players
   end
 
+  def retrieve_user_move
```

Comment:
This is fine for now, but ultimately I think it will be easier to test and read more clearly if it takes the current player as an argument. Specifically, I'm talking about how your test says `it "asks the current player for their move"`, and goes on to assert that it asks `X` for the move. It's unclear why it would be X's turn though. I imagine you'll change this a bit when you work on an upcoming story.

Reply:
+1 to @aaizuss 's comments. Agreed